tests/test_focusAnalysis.py

```diff
@@ -23,11 +23,23 @@
 from typing import Iterable
 
 import matplotlib as mpl
+import matplotlib.pyplot as plt
 
 import lsst.utils.tests
 
 mpl.use("Agg")
 
+
+def custom_show(*args, **kwargs):
+    pass
+
+
+# Override the show function with our custom one
+# to avoid getting the next Warning:
+# UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
+# plt.show()
+plt.show = custom_show
```

Comment:
I wonder if the right solution isn't just to remove the calls to `plt.show()` to be honest, I doubt that's ever actually used interactively.

Reply:
So we should remove the show inside the code? The show() is not part from the test but from the function in the library. I guess you mean returning the matplotlib Plot and then the uplayer call the show() from it, (but that is a big change)

Reply:
As I put on the other PR, yes, I think that's fine to remove, nobody will be using this interactively.